modules/wsl-conf.nix

```diff
@@ -0,0 +1,90 @@
+{ lib, pkgs, config, ... }:
+with builtins; with lib; {
+
+  imports = [
+    (mkRenamedOptionModule [ "wsl" "automountPath" ] [ "wsl" "wslConf" "automount" "root" ])
+    (mkRenamedOptionModule [ "wsl" "automountOptions" ] [ "wsl" "wslConf" "automount" "options" ])
+  ];
+
+  # See https://learn.microsoft.com/en-us/windows/wsl/wsl-config#configuration-settings-for-wslconf for all options
+  options.wsl.wslConf = with types; {
+    automount = {
+      enabled = mkOption {
+        type = bool;
+        default = true;
+        description = "Automatically mount windows drives under ${config.wsl.wslConf.automount.root}";
+      };
+      mountFsTab = mkOption {
+        type = bool;
+        default = false;
+        description = "Mount entries from /etc/fstab through WSL. You should probably leave this on false, because systemd will mount those for you.";
+      };
+      root = mkOption {
+        type = str;
+        default = "/mnt/";
+        description = "The directory under which to mount windows drives.";
+      };
+      options = mkOption {
+        type = commas; # comma-separated strings
+        default = "metadata,uid=1000,gid=100";
+        description = "Comma-separated list of mount options that should be used for mounting windows drives.";
+      };
+    };
+    boot = {
+      command = mkOption {
+        type = str;
+        default = "";
+        description = "A command to run when the distro is started.";
+      };
+      systemd = mkOption {
+        type = bool;
+        default = false;
+        description = "Use systemd as init. There's no need to enable this manually, use the wsl.nativeSystemd option instead";
+      };
+    };
+    interop = {
+      enabled = mkOption {
+        type = bool;
+        default = true;
+        description = "Support running Windows binaries from the linux shell.";
+      };
+      appendWindowsPath = mkOption {
+        type = bool;
```

Comment:
This is currently a string which contains the boolean.

```
error: A definition for option `wsl.wslConf.interop.appendWindowsPath' is not of type `boolean'. Definition values:
       - In `/nix/store/byrkin8wmr9hbzqhnsfa01bj2g7xij81-source/common/modules.nix': "false"
```

Reply:
Where is that string? Was it an old default config? The current default configuration.nix as well as my flake config build just fine

Reply:
It is some option I have currently set but I no longer know why and from where I got it. Also can't find any reference in code.

Reply:
I think I now know the root cause of this: before wslConf was a freeform setting that accepted a lot more than it should. Now that the options are properly typed errors are properly surfaced and you can no longer set an identical string to an otherwise boolean option.